src/Raven.Client/Documents/Commands/Batches/BatchOptions.cs

```diff
@@ -25,4 +27,24 @@ public class ReplicationBatchOptions
         public bool Majority { get; set; }
         public bool ThrowOnTimeoutInWaitForReplicas { get; set; }
     }
+
+    public class ShardedBatchOptions
+    {
+        internal static readonly ShardedBatchOptions NonTransactionalMultiBucket = new() { BatchBehavior = ShardedBatchBehavior.NonTransactionalMultiBucket };
+
+        internal static readonly ShardedBatchOptions TransactionalSingleBucketOnly = new() { BatchBehavior = ShardedBatchBehavior.TransactionalSingleBucketOnly };
+
+        public ShardedBatchBehavior BatchBehavior { get; set; }
+
+        internal static ShardedBatchOptions For(ShardedBatchBehavior behavior)
+        {
+            return behavior switch
+            {
+                ShardedBatchBehavior.Default => null,
+                ShardedBatchBehavior.TransactionalSingleBucketOnly => TransactionalSingleBucketOnly,
+                ShardedBatchBehavior.NonTransactionalMultiBucket => NonTransactionalMultiBucket,
+                _ => throw new ArgumentOutOfRangeException(nameof(behavior), behavior, null)
+            };
```

Comment:
I'm not fan of the names here... I suggest to name this `TransactionalGranularity`.
`TransactionalGranularity.Default` / `None`
`TransactionalGranularity.SingleBucket`
`TransactionalGranularity.SingleShard`

Reply:
we can even add `TransactionalGranularity.MultipleShards` and use the cluster-wide tx.

Reply:
We don't *have* single shard granularity. Nor can you rely on that.

And cluster wide transaction is something that we would like users to explicitly use